profiles/utilities.py

```diff
@@ -15,6 +16,20 @@
 from profiles.exceptions import SchemaNotFound
 
 
+def catch_exceptions(logger: Logger) -> Callable[..., Any]:
+    def outer_wrapper(func: Callable[..., Any]) -> Callable[..., Any]:
+        @wraps(func)
+        def wrapper(*args, **kwargs) -> Any:
+            try:
+                return func(*args, **kwargs)
+            except BaseException as exception:
```

Comment:
Catching `BaseException` will also catch system exiting exceptions. Blindly catching these can be dangerous, could we not use `Exception` so at least its just catching non-system exit exceptions?

Reply:
Would any non-`Exception` crash the application and cause it to restart?

Reply:
Hang on, misread. So any exception that doesn't extend `Exception` is definitely going to be a shutdown? In that case, we can ignore those.